packages/core/src/utilities/getRandomSampleFromArray.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Returns a random sample of elements from an array.
+ * @param array - The input array.
+ * @param size - The number of elements to sample.
+ * @returns A new array containing the randomly sampled elements.
+ */
+export function getRandomSampleFromArray(array, size) {
+  const result = [];
+  const len = array.length;
+  const set = new Set(); // Use a Set to track selected indices for O(1) lookups
+
+  while (result.length < size) {
```

Comment:
If `size` is greater than `array.length` this function will never return and if the size is greater than 50% the number of "hits" (when the set already contains the number) will increase considerably and we will never know when it will return.

I would implement it in a more predictable way. Example, sort using random + slice N first elements.

Reply:
There is a method in lodash that do something similar:
- clone the array
- loop through all elements and choose a random index to switch with the current element
- take the first N elements

`sampleSize`
https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L9791-L9799

`arraySampleSize`
https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L2430C14-L2432

`shuffleSelf`
https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L6711-L6726

Reply:
I saw the size being limited by the caller in `projectTo2D.ts` (`Math.min(50, polyline.length))  but I think it should be here otherwise any other function that calls this method will have to do the same to avoid an infinity loop.